xonsh/shell.py

```diff
@@ -87,7 +89,12 @@ def __init__(self, ctx=None, shell_type=None, config=None, rc=None,
         self._init_environ(ctx, config, rc,
                            kwargs.get('scriptcache', True),
                            kwargs.get('cacheall', False))
+
         env = builtins.__xonsh_env__
+        # build history backend before creating shell
+        builtins.__xonsh_history__ = xhm.construct_history(
+            env=env.detype(), ts=[time.time(), None], locked=True)
```

Comment:
hmm this might be OK for this PR.  We really need a pre-shell/post-load event, I think.